### DISCUSSION TITLE
DataGrid pagination_click event

### DISCUSSION THREAD

**Discussion by stenci** [anvilistas/anvil-extras#528] [Q&A]
I have a problem with the `data_grid.set_event_handler('pagination_click', self.pagination_click)`.

I am able to get my handler to run, but I can't prevent the default one from running.

I would like to `e.preventDefault()`, so I can add items to the list when we are approaching the end before showing the next page.

I tried to `return True` on my event handler, but didn't help. My handler runs and the page is loaded anyway.

Any idea about how to get `e.preventDefault()` to work, or something equivalent?

**Comment by s-cork** [comment]
`e.preventDefault()` is typically used to prevent default browser behaviour.
But anvil's own event listeners aren't going to respect that.

Instead you'd need to be able to call `e.stopPropagation()` to stop the event from propagating to anvil's handlers.

Locally you can try copying the augment module to try out some changes
Use your local version rather than the anvil-extras one

Then adding this to line 215 to be

```python
def _wrap_js_event(handler):
    def wrapper(e):
        handler(original_event=e)

    return wrapper


```

Then in your handler

```python
    def my_data_grid_click_handler(self, **event_args):
        original_event = event_args["original_event"]
        original_event.stopPropagation() # stop this event from propagating
        # original_event.stopImmediatePropagation() # might be required try this if the above doesnt work
        ... # rest of function
```

If that works we can look to merge it upstream

**Comment by stenci** [comment]
Hi Stu,
Thank you for the reply, but I tried with your suggestions and it didn't work.

So I am giving up with managing the pagination events.

Instead I created a little `DynamicIterator` class that works with a DataGrid just like the `SearchIterator`. It lazily loads items in chunks, which is what I wanted to do with the pagination events anyway.

I will publish this little class in the forum in a few days, after a little real world testing.
